common/models/user.js

```diff
@@ -544,7 +544,11 @@ module.exports = function(User) {
     this.settings.ttl = this.settings.ttl || DEFAULT_TTL;
 
     UserModel.setter.password = function(plain) {
-      if (plain.indexOf('$2a$') === 0 && plain.length === 60) {
+      var indexOf = String.prototype.indexOf;
+      if (!plain) {
+        return;
+      }
+      if (indexOf.call(plain, '$2a$') === 0 && plain.length === 60) {
```

Comment:
@raymondfeng I don't think this really covers our use case. Most of our users coming in through github/twitter auth, they don't set passwords. We need to make sure users who do not have passwords do not crash the site just by logging in. 

I can get ride of the indexOf reflection and just set it to this:
```js
    UserModel.setter.password = function(plain) {
      if (!plain) {
        return;
      }
      if (plain.indexOf('$2a$') === 0 && plain.length === 60) {
        // The password is already hashed. It can be the case
        // when the instance is loaded from DB
        this.$password = plain;
      } else {
        this.$password = this.constructor.hashPassword(plain);
      }
    };
```

This prevents crashes when plain is undefined, which is the use case I am trying to solve, but won't do anything for the case when plain is a number, which I don't really need right now. 

@raymondfeng How does that sound?